### PR TITLE
Add+delete product

### LINKED ISSUE #3: Adding Products to a Wishlist
**As a** user
**I need** to be able to add products to a particular wishlist
**So that** I can organize my sought-after products
   
 ### Details and Assumptions
 * At most one instance of a product may belong to a particular wishlist
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given that the service is available
 When the user attempts to add a product to a particular wishlist
 Then one of two things should happen:
 1.  If no wishlist with the given name exists, the addition must fail (404_NOT_FOUND)
 2.  Else: 
           1.  If the product already belongs to this wishlist, the addition must fail (409_CONFLICT)
           2.  Else, an entry into the database must be made thereby indicating successful addition (201_CREATED)

```

### LINKED ISSUE #12: Removing Products from a Wishlist
**As a** user
**I need** to be able to able to remove products from a particular wishlist
**So that** I can no longer treat it as a sought-after product  
   
 ### Details and Assumptions
 * The product to be removed must belong to the wishlist under consideration
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given that the service is available
 When the user attempts to remove a product from a particular wishlist
 Then one of two things should happen:
 1.  If no wishlist with the given name exists, the removal must fail (404_NOT_FOUND)
 2.  Else:
           1.  If the product does not belong to this wishlist, the removal must fail (404_NOT_FOUND)
           2.  Else, the entry corresponding to this product-wishlist mapping must be removed from the database thereby indicating successful removal (204_NO_CONTENT)

```

### PR DESCRIPTION
**Pull request by sathvikabm**
Adding and removing product from wishlist 

This code doesnot have Base class model

### CODE REVIEW

File: tests/factories.py
Comment (sohailhodarkar) [inline tests/factories.py:1]: Please change line 57 to ```wishlist_id = None```. The rest is perfect

File: service/routes.py
Comment (sohailhodarkar) [inline service/routes.py:1]: This is great, but I have a couple of questions/observations here:

1. Retrieve an "account" should be renamed to retrieve a "wishlist"
2. Why have you added code for 1 (retrieving a wishlist)? I thought somebody was working on that story. Did you need it elsewhere? Same goes for listing the products in a wishlist. I understand that you needed "read a product" for your tests.
3. The way we modeled our "add a product" feature, was that a product cannot be added to the same wishlist multiple times. This code does not address that. I have a possible solution for that, which I will implement after merging.

Please answer the questions/make changes, and we are good to merge!

File: tests/factories.py
Comment (sathvikabm) [reply tests/factories.py:1]: Done

File: service/routes.py
Comment (sathvikabm) [reply service/routes.py:1]: 2. So when we run thunderbolt, you input the values right,
Now it is so much easier if the functions are already defined to check what are all the wishlists present and then add the product
Else its so difficult to keep track of everything

One more reason was as you stated when the path is not defined for GET, POST 
we cant add or delete hence the read function

you can add your implementation for same wishlist multiple times

**Review by sohailhodarkar — APPROVED** [review]
Merge conflicts need to be fixed. Otherwise, good to merge!

**Comment by sohailhodarkar** [comment]
Merge conflicts resolved. Good to merge!

**Pull request merged**